### PR TITLE
Validate search

### PR DESCRIPTION
**Pull request by marius-patrascu**
Added validation for the search on home page

### CODE REVIEW

File: src/components/MainSearch/index.tsx [hunk 1]
```diff
@@ -12,12 +12,12 @@ export function MainSearch() {
...
 
-  const [types, setTypes]: [any, any] = useState([]);
-  const [q, setQ]: [any, any] = useState('');
+  const [types, setTypes]: [any, any] = useState({value: [], isPristine: true});
+  const [q, setQ]: [any, any] = useState({value: '', isPristine: true});
 
   const handleKeyPress = (event: any) => {
-    if(event.key === 'Enter'){
-      navigate(`${ASSET_CATALOG_PATH}?q=${q}&types=${types.join(',')}`)
+    if (event.key === 'Enter') {
```
Comment (sarbull) [inline src/components/MainSearch/index.tsx:19]:
```js
// check if user input is valid before redirect with invalid input
if (event.key === 'Enter' && (q.isValid && types.isValid)) {
```

File: src/components/MainSearch/index.tsx [hunk 2]
```diff
@@ -12,12 +12,12 @@ export function MainSearch() {
     typesData: []
   });
 
-  const [types, setTypes]: [any, any] = useState([]);
-  const [q, setQ]: [any, any] = useState('');
+  const [types, setTypes]: [any, any] = useState({value: [], isPristine: true});
```
Comment (sarbull) [inline src/components/MainSearch/index.tsx:15]: use same pattern
```js
{ value: [], isPristine: true, isValid: false }
```

File: src/components/MainSearch/index.tsx [hunk 3]
```diff
@@ -12,12 +12,12 @@ export function MainSearch() {
...
+  const [q, setQ]: [any, any] = useState({value: '', isPristine: true});
```
Comment (sarbull) [inline src/components/MainSearch/index.tsx:16]: use same pattern
```js
{ value: '', isPristine: true, isValid: false }
```

File: src/components/MainSearch/index.tsx [hunk 4]
```diff
@@ -38,26 +38,28 @@ export function MainSearch() {
...
-                size="xs"
-                value={q}
-                disabled={typesData.isLoading}
-                onKeyPress={handleKeyPress}
-                onChange={(event: any) => setQ(event.currentTarget.value)}
-                placeholder="Search terms"
-                rightSectionWidth={42} />
+                 value={types.value}
+                 disabled={typesData.isLoading}
+                 onChange={(value: any) => setTypes({value: [...value]})}
```
Comment (sarbull) [inline src/components/MainSearch/index.tsx:43]: use same pattern
```js
onChange={(value: any) => setTypes({
  value: [...value],
  isPristine: false,
  isValid: !isArrayEmpty([...value])
}
)}
```

File: src/components/MainSearch/index.tsx [hunk 5]
```diff
@@ -38,26 +38,28 @@ export function MainSearch() {
...
+
+    <TextInput style={{width: '30%'}}
+               icon={<Search size={18}/>}
+               radius="sm"
+               size="xs"
+               value={q.value}
+               disabled={typesData.isLoading}
+               error={!q.isPristine && !isStringLonger(q.value, QUERY_MIN_LENGTH) ? 'Query must be at least ' + QUERY_MIN_LENGTH + ' characters' : ''}
+               onKeyPress={handleKeyPress}
+               onChange={(event: any) => setQ({value: event.currentTarget.value})}
```
Comment (sarbull) [inline src/components/MainSearch/index.tsx:61]: use same pattern
```js
onChange={(value: any) => setQ({
  value: event.currentTarget.value,
  isPristine: false,
  isValid: !isStringLonger(event.currentTarget.value, QUERY_MIN_LENGTH)
}
)}
```

File: src/components/MainSearch/index.tsx [hunk 6]
```diff
@@ -38,26 +38,28 @@ export function MainSearch() {
...
+                   typesData.isLoading ? <Loader size={22}/> : <></>
+                 }/>
+
+    <TextInput style={{width: '30%'}}
+               icon={<Search size={18}/>}
+               radius="sm"
+               size="xs"
+               value={q.value}
+               disabled={typesData.isLoading}
+               error={!q.isPristine && !isStringLonger(q.value, QUERY_MIN_LENGTH) ? 'Query must be at least ' + QUERY_MIN_LENGTH + ' characters' : ''}
```
Comment (sarbull) [inline src/components/MainSearch/index.tsx:59]:
```js
error={ !(!q.isPristine && q.isValid) ? `Query must be at least ${QUERY_MIN_LENGTH} characters' : ''}
```

File: src/components/MainSearch/index.tsx [hunk 7]
```diff
@@ -38,26 +38,28 @@ export function MainSearch() {
...
-                value={q}
-                disabled={typesData.isLoading}
-                onKeyPress={handleKeyPress}
-                onChange={(event: any) => setQ(event.currentTarget.value)}
-                placeholder="Search terms"
-                rightSectionWidth={42} />
+                 value={types.value}
+                 disabled={typesData.isLoading}
+                 onChange={(value: any) => setTypes({value: [...value]})}
+                 error={!types.isPristine && isArrayEmpty(types.value) ? 'At least one type has to be selected' : ''}
```
Comment (sarbull) [inline src/components/MainSearch/index.tsx:44]:
```js
error={ !(!types.isPristine && types.isValid) ? 'At least one type has to be selected' : ''}
```

**Review by sarbull — CHANGES_REQUESTED** [review]

File: dist/esm/components/About/index.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import { jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from "react/jsx-runtime";
```
[the remark this answers is not in the document]
Comment (marius-patrascu) [reply dist/esm/components/About/index.js]: is removed

File: src/components/MainSearch/index.tsx (hunk 1 above)
Comment (marius-patrascu) [reply src/components/MainSearch/index.tsx:19]: done

File: src/components/MainSearch/index.tsx (hunk 3 above)
Comment (marius-patrascu) [reply src/components/MainSearch/index.tsx:16]: done

File: src/components/MainSearch/index.tsx (hunk 4 above)
Comment (marius-patrascu) [reply src/components/MainSearch/index.tsx:43]: done

File: src/components/MainSearch/index.tsx (hunk 5 above)
Comment (marius-patrascu) [reply src/components/MainSearch/index.tsx:61]: done

File: src/components/MainSearch/index.tsx (hunk 6 above)
Comment (marius-patrascu) [reply src/components/MainSearch/index.tsx:59]: done

File: src/components/MainSearch/index.tsx (hunk 7 above)
Comment (marius-patrascu) [reply src/components/MainSearch/index.tsx:44]: done

**Review by sarbull — APPROVED** [review]

**Pull request merged**